### PR TITLE
CFM-760: Create block to display 5 groups

### PR DESCRIPTION
**Pull request by DavidHernandez**
Increments amount of groups shown on the "my groups" homepage block.

Replaces JS link to toggle the list of groups with a link to the groups listing page filtered.

Issue: https://issuetracker.amplexor.com/jira/browse/CFM-760

my-groups

### CODE REVIEW

**Comment by DavidHernandez** [comment]
## Merge https://github.com/capacity4dev/capacity4more/pull/1139 first.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc [hunk 1]
```diff
@@ -19,57 +19,79 @@ $plugin = array(
...
+/**
+ * Returns the list of groups of an user, grouped by the membership type.
+ *
+ * @param object $account.
+ *   User object. Defaults global user if NULL.
+ *
+ * @returns mixed|null
+ *   List of groups or NULL if the user has no groups.
+ */
+function c4m_features_homepage_get_user_groups_grouped_by_membership($account = NULL) {
```
Comment (soniCaH) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: Functionality seems nowhere used.
Please also make sure owner is on top > followed by admin > member > membership requested.

Ref:

show on the home page
- groups I own ("owner")
- groups I manage ("admnistrator")
- groups I'm member of ("member")
- groups I requested membership for ("membership requested")

"my archived groups" do not need to be shown on home page
ordering

1st: owner > administrator > member > membership requested
2nd: alphabetical

**Comment by ordavidil** [comment]
@DavidHernandez 
please add new screenshots ...

**Comment by DavidHernandez** [comment]
Desktop:
my-groups-desktop
Tablet:
my-groups-tablet
Mobile:
my-groups-mobile

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc [hunk 2]
```diff
@@ -20,56 +20,110 @@ $plugin = array(
  */
 function c4m_features_homepage_my_groups_content_type_render($subtype, $conf, $args, $context) {
   global $user;
+  $groups = c4m_features_homepage_get_user_groups_grouped_by_membership($user, array('group'));
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: @DavidHernandez 
Method call uses 2 parameters but method signature uses 1.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc [hunk 3]
```diff
@@ -20,56 +20,110 @@ $plugin = array(
...
-  $first_groups = '';
-  $extra_groups = '';
-
-  // Get All user's groups (We get them sorted by id).
-  $groups = og_get_groups_by_user($account, NULL, TRUE);
-
-  if (empty($groups['node'])) {
-    return FALSE;
+  if (empty($groups)) {
+    return;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: Missing return argument.

**Pull request merged**